src/Nethermind/Nethermind.Consensus/Withdrawals/WithdrawalProcessor.cs

```diff
@@ -31,8 +31,9 @@ public void ProcessWithdrawals(Block block, IReleaseSpec spec)
 
         if (block.Withdrawals is not null)
         {
-            foreach (var withdrawal in block.Withdrawals)
+            for (int i = 0; i < block.Withdrawals.Length; i++)
```

Comment:
Take array to local if using for with length; or Jit won't eliminate the bounds checks as the array length could change
```csharp
var withdrawals = block.Withdrawals;
for (int i = 0; i < withdrawals.Length; i++)
    Withdrawal withdrawal = withdrawals[i];
```